### PR TITLE
executor: skip chunk evaluation in splitIntoGroups when unnecessary

### PR DESCRIPTION
**Pull request by krzysztofpioro**
### What problem does this PR solve?

When `firstRowDatums` are equal to `lastRowDatums` then there is no need to evaluate chunk. 

### What is changed and how it works?

Extract getting `firstRowDatums` and `lastRowDatums` from `evalGroupItemsAndResolveGroups` and call  `evalGroupItemsAndResolveGroups` only when `firstGroupKey` is different than `lastGroupKey`.

Benchmark for [BenchmarkAggGroupByNDV](https://github.com/pingcap/tidb/blob/a9dc72cf25589bc00260f1af4e5a17dac0f66474/executor/benchmark_test.go#L383):
```
name                                                                                                           old time/op  new time/op  delta
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:10,_hasDistinct:false,_rows:10000000,_concurrency:4)-4         163ms ± 3%   104ms ± 5%  -35.86%  (p=0.008 n=5+5)
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:100,_hasDistinct:false,_rows:10000000,_concurrency:4)-4        161ms ± 1%   104ms ± 1%  -35.50%  (p=0.008 n=5+5)
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:1000,_hasDistinct:false,_rows:10000000,_concurrency:4)-4       162ms ± 1%   108ms ± 0%  -32.95%  (p=0.008 n=5+5)
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:10000,_hasDistinct:false,_rows:10000000,_concurrency:4)-4      162ms ± 0%   158ms ± 2%   -2.50%  (p=0.016 n=5+5)
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:100000,_hasDistinct:false,_rows:10000000,_concurrency:4)-4     168ms ± 1%   163ms ± 1%   -2.76%  (p=0.008 n=5+5)
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:1000000,_hasDistinct:false,_rows:10000000,_concurrency:4)-4    225ms ± 1%   227ms ± 3%     ~     (p=0.690 n=5+5)
AggGroupByNDV/(execType:stream,_aggFunc:sum,_ndv:10000000,_hasDistinct:false,_rows:10000000,_concurrency:4)-4   494ms ± 2%   487ms ± 3%     ~     (p=0.222 n=5+5)

```

### Check List 

Tests 

- Unit test

### Release note 

- skip chunk evaluation in splitIntoGroups when unnecessary

### CODE REVIEW

**Comment by krzysztofpioro** [comment]
@SunRunAway PTAL

File: executor/aggregate.go [hunk 1]
```diff
@@ -1080,6 +1072,27 @@ func (e *vecGroupChecker) splitIntoGroups(chk *chunk.Chunk) (isFirstGroupSameAsP
 	}
 	copy(e.lastGroupKeyOfPrevChk, e.lastGroupKey)
 
+	if bytes.Equal(e.firstGroupKey, e.lastGroupKey) {
+		e.groupOffset = append(e.groupOffset, numRows)
+		e.groupCount = len(e.groupOffset)
```
Comment (Reminiscent) [inline executor/aggregate.go:1077]: We can directly set `e.groupCount = 1`.

**Pull request merged**